### ISSUE TITLE
Curry

### ISSUE THREAD

**Issue by EggDice** [conorhastings/match-default#1]
Hi!

I love the simplicity of this lib, also I think it have a great use-case with redux :)
Would it make sense for you to have a curry functionality?
Something like this:

``` js
const matcher = match({
  a: () => 'apple',
  _: () => 'default'
})
console.log(matcher(apple)) // apple
```

If it makes, I would happily send a pull request!

**Comment by conorhastings** [comment]
@EggDice definitely, great idea! would happily accept a PR.

**Comment by conorhastings** [comment]
@EggDice finally got around to this, added https://github.com/conorhastings/match-default/pull/2 here, let me know if this works for you!

published as `1.2.0`
